Automattic/woocommerce-payments
# Add multi-currency enablement check to WooPay session handler

Fixes #7329 

#### Changes proposed in this Pull Request

Title: Add multi-currency enablement check to WooPay session handler


Description: Adding multi-currency enablement check to WooPay session handler to avoid unwanted multi-currency behavior for merchants using WooPay that have disabled multi-currency. @bborman22 describes the scenario here: https://github.com/Automattic/woocommerce-payments/issues/7329#issuecomment-1747467641


#### Testing instructions
(Instructions pulled from https://github.com/Automattic/woocommerce-payments/pull/7055#issue-1863965079)

1. Ensure WooPay is eligible and enabled.
2. Enable Multi-Currency + the currency switcher from WC > Settings > Multi-Currency.
3. Enable multiple currencies, for example: USD, JPY, EUR.
4. As a guest shopper use the currency switcher to change the currency to JPY.
5. Navigate to the checkout page and enter a valid WooPay email + OTP.
6. From the WooPay checkout, ensure the currency is displayed correctly.
7. In a separate tab, go to the merchant store and change the currency to, "USD", for example.
8. Refresh the WooPay checkout tab and ensure the currency, totals and currency symbol are set to USD.
9. Checkout with WooPay.
10. Ensure the charge is correctly processed for the chosen currency.
11. Logout from WooPay and repeat steps 4-10 as a logged in shopper.
12. Disable multi-currency and repeat steps 4-11




**Post merge**
